src/rules/require-top-level-describe.ts

```diff
@@ -0,0 +1,50 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+import { createRule, isDescribe, isHook, isTestCase } from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Prevents test cases and hooks to be outside of a describe block',
+      recommended: false,
+    },
+    messages: {
+      unexpectedTestCase: 'All test cases must be wrapped in a describe block.',
+      unexpectedHook: 'All hooks must be wrapped in a describe block.',
+    },
+    type: 'suggestion',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    let numberOfDescribeBlocks = 0;
+    return {
+      CallExpression(node) {
+        if (isDescribe(node)) {
+          numberOfDescribeBlocks++;
+          return;
+        }
+
+        if (numberOfDescribeBlocks === 0) {
+          if (isTestCase(node)) {
+            context.report({ node, messageId: 'unexpectedTestCase' });
+            return;
+          }
+
+          if (isHook(node)) {
```

Comment:
CI complains that this is not covered. Maybe the `else` branch? A test with a `CallExpression` inside a `describe` that's neither a test nor a hook should probably hit it?

Reply:
Indeed! I added a dummy test to fix it: 

```js
'foo()',
```

But thinking of it, should we trigger a warning if the user uses a `CallExpression` that's neither a test or a hook or can we leave it as is?

Reply:
no, that's fine, I think. Jest itself uses it to e.g. skip tests on windows, or run some other sort of helper 🙂